### PR TITLE
Ssl assert fingerprint

### LINKED ISSUE #164: ✨ Implement an option for disable SSL Certificate Verify
Hello, 

We are trying to deploy it without SSL signed on ElasticSearch side, and it doesn't seem to work without SSL Signed.

Please find below the error we have : 

```JSON
{
    "@timestamp": "2022-09-29T11:54:37.848Z",
    "log.level": "warning",
    "message": "elasticsearch shipper",
    "_id": "f5d4f08ae9-000000000000",
    "ecs": {
        "version": "1.6.0"
    },
    "error": "ConnectionError([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)) caused by: SSLError([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129))",
    "log": {
        "logger": "root",
        "origin": {
            "file": {
                "line": 167,
                "name": "es.py"
            },
            "function": "_handle_outcome"
        },
        "original": "elasticsearch shipper"
    },
    "process": {
        "name": "MainProcess",
        "pid": 8,
        "thread": {
            "id": 140115916543808,
            "name": "MainThread"
        }
    }
}
```

Would it be possible to add an option into the `config.yaml` file to handle no certificate verification (We use auto-signed certificate).

Many thanks for the :sparkles: ^^ 

Hints Reference : 
https://github.com/elastic/elastic-serverless-forwarder/blob/8fd2d8af994bf91536f04ff3ccf24335b759e3ee/shippers/es.py#L108

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement
## What does this PR do?
Add support to connect to ES with self-signed certificate without completely disable SSL verification (relying at least on self-singed certificate fingerprint validation)

## Why is it important?
Unlock usage of ESF connecting to ES with self-signed  certificate

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.md`

## Author's Checklist

- [ ]

## Related issues
Closes #164

-

### CODE REVIEW

**Review by tommyers-elastic — APPROVED** [review]
couple of small things - the most important being the `verify_certs` option. but overall happy with the addition of this parameter - thanks!

File: docs/README-AWS.md [hunk 1]
```diff
@@ -373,6 +378,7 @@ For `elasticsearch` the following arguments are supported:
...
   * `args.batch_max_actions`: Maximum number of actions to send in a single bulk request. Default value: 500
   * `args.batch_max_bytes`: Maximum size in bytes to send in a single bulk request. Default value: 10485760 (10MB)
+  * `args.ssl_assert_fingerprint`: SSL fingerprint for self-signed SSL certificate on HTTP transport
```
[the remark this answers is not in the document]
Comment (aspacca) [reply docs/README-AWS.md:381]: good catch!
it's `HTTPS`

File: share/config.py [hunk 2]
```diff
@@ -63,6 +64,7 @@ def __init__(
         self.tags = tags
         self.batch_max_actions = batch_max_actions
         self.batch_max_bytes = batch_max_bytes
+        self.ssl_assert_fingerprint = ssl_assert_fingerprint
```
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply share/config.py:67]: to use the getter it should read `self._ssl_assert_fingerprint = ssl_assert_fingerprint`. the code as it stands replaces the property with the actual value.

[the remark this answers is not in the document]
Comment (aspacca) [reply share/config.py:67]: `self._ssl_assert_fingerprint = ssl_assert_fingerprint` will bypass the setter :)

File: shippers/es.py [hunk 3]
```diff
@@ -89,6 +90,10 @@ def __init__(
         else:
             raise ValueError("You must provide one between username and password or api_key")
 
+        if ssl_assert_fingerprint:
+            es_client_kwargs["verify_certs"] = False
```
[the remark this answers is not in the document]
Comment (aspacca) [reply shippers/es.py:94]: it turned out that passing `ssl_assert_fingerprint` assert the fingerprint and still verifies the certs
in order to assert only the fingerprint without certs chain verification you have to set `verify_certs=False`

with this, if you pass the wrong fingerprint the connection will fail. I've asked for confirmation but this behaviour is reflected by my local tests

I could add a test scenario for the wrong fingerprint

[the remark this answers is not in the document]
Comment (aspacca) [reply shippers/es.py:94]: > I could add a test scenario for the wrong fingerprint

done

**Pull request merged**